core/templates/components/forms/custom-forms-directives/image-uploader-modal.component.html

```diff
@@ -0,0 +1,221 @@
+<div class="oppia-image-uploader-modal-container">
+  <div class="modal-header">
+    <h3>{{ 'I18N_IMAGE_UPLOADER_MODAL_UPLOAD_HEADING' | translate:{ imageName: imageName } }}</h3>
+  </div>
+
+  <div class="modal-body oppia-image-uploader-container">
+    <div *ngIf="(imageName==='Thumbnail' && !uploadedImage && !thumbnailImageDataUrl)">
```

Comment:
I see too much logic in html here. Let's introduce functions in TS that have descriptive names. It's not clear what is "(imageName==='Thumbnail' && !uploadedImage && !thumbnailImageDataUrl)". Same goes for other non-trivial if-s

Reply:
yes make sense. I have created isThumbnail() and imageNotUploaded() functions.  we show this when imageName is thumbnail and no image is uploaded.